responses/__init__.pyi

```diff
@@ -0,0 +1,285 @@
+) -> Callable[..., Any]: ...
+
+class Call(NamedTuple):
+    request: PreparedRequest
+    response: Any
+
+_Body = Union[str, BaseException, "Response", BufferedReader, bytes]
+
+class CallList(Sequence[Call], Sized):
+    def __init__(self) -> None: ...
+    def __iter__(self) -> Iterator[Call]: ...
+    def __len__(self) -> int: ...
+    def __getitem__(self, idx: int) -> Call: ...  # type: ignore [override]
+    def add(self, request: PreparedRequest, response: _Body) -> None: ...
+    def reset(self) -> None: ...
+
+class BaseResponse:
+    content_type: Optional[str] = ...
+    headers: Optional[Mapping[str, str]] = ...
+    stream: bool = ...
+    method: Any = ...
+    url: Any = ...
+    match_querystring: Any = ...
+    match: List[Any] = ...
+    call_count: int = ...
+    def __init__(
+        self,
+        method: str,
+        url: Union[Pattern[str], str],
+        match_querystring: Union[bool, object] = ...,
+        match: List[Callable[..., Any]] = ...,
+    ) -> None: ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __ne__(self, other: Any) -> bool: ...
+    def _body_matches(
+        self, match: List[Callable[..., Any]], request_body: Optional[Union[bytes, str]]
+    ) -> bool: ...
+    def _should_match_querystring(
+        self, match_querystring_argument: Union[bool, object]
+    ) -> bool: ...
+    def _url_matches(
+        self, url: Union[Pattern[str], str], other: str, match_querystring: bool = ...
+    ) -> bool: ...
+    def _url_matches_strict(self, url: str, other: str) -> bool: ...
+    def get_headers(self) -> HTTPHeaderDict: ...  # type: ignore [no-any-unimported]
+    def get_response(self, request: PreparedRequest) -> None: ...
+    def matches(self, request: PreparedRequest) -> Tuple[bool, str]: ...
+
+class Response(BaseResponse):
+    body: _Body = ...
+    status: int = ...
+    headers: Optional[Mapping[str, str]] = ...
+    stream: bool = ...
+    content_type: Optional[str] = ...
+    def __init__(
+        self,
+        method: str,
+        url: Union[Pattern[str], str],
+        body: _Body = ...,
+        json: Optional[Any] = ...,
+        status: int = ...,
+        headers: Optional[Mapping[str, str]] = ...,
+        stream: bool = ...,
+        content_type: Optional[str] = ...,
+        match_querystring: bool = ...,
+        match: List[Any] = ...,
+    ) -> None: ...
+    def get_response(  # type: ignore [override]
+        self, request: PreparedRequest
+    ) -> HTTPResponse: ...
+
+class CallbackResponse(BaseResponse):
+    callback: Callable[[Any], Any] = ...
+    stream: bool = ...
+    content_type: Optional[str] = ...
+    def __init__(
+        self,
+        method: str,
+        url: Union[Pattern[str], str],
+        callback: Callable[[Any], Any],
+        stream: bool = ...,
+        content_type: Optional[str] = ...,
+        match_querystring: bool = ...,
+        match: List[Any] = ...,
+    ) -> None: ...
+    def get_response(  # type: ignore [override]
```

Comment:
I used the following config when developing, not sure if you want mypy running or not in CI


```ini
[mypy]
show_column_numbers=True
show_error_codes = True

disallow_any_unimported=True
disallow_any_expr=False
disallow_any_decorated=True
disallow_any_explicit=False
disallow_any_generics=True
disallow_subclassing_any=True

disallow_untyped_calls=True
disallow_untyped_defs=True
disallow_incomplete_defs=True
check_untyped_defs=True
disallow_untyped_decorators=True

no_implicit_optional=True
strict_optional=True

warn_redundant_casts=True
warn_unused_ignores=True
warn_no_return=True
warn_return_any=False
warn_unreachable=False

strict_equality=True

ignore_missing_imports=False

[mypy-urllib3.*]
ignore_missing_imports=True

[mypy-test_responses]
disallow_untyped_defs=False
disallow_untyped_calls=False
disallow_any_decorated=False
```

Reply:
Having mypy run as part of CI would be ideal. I can take care of that though.

Reply:
How were you running mypy locally? When I ran it with this configuration file there were ~380 errors from the test_responses module.

Reply:
not sure what command I was running, but were you using the config above?

```
[mypy-test_responses]
disallow_untyped_defs=False
disallow_untyped_calls=False
disallow_any_decorated=False
```

should ignore most of the errors until the tests module is fully typed